packages/core/src/react-integration/provider/BackupBoundary.tsx

```diff
@@ -1,15 +1,26 @@
 import React, { Suspense, memo, useMemo, version } from 'react';
 
 /* istanbul ignore next  */
-const NoSuspense =
-  (version.startsWith('16') || version.startsWith('17')) &&
-  typeof window === 'undefined';
+const LegacyReact = version.startsWith('16') || version.startsWith('17');
+/* istanbul ignore next  */
+const SSR = typeof document === 'undefined';
```

Comment:
i'd probably use `window` but this should be fine too

Reply:
I figured since I was actually using document for the condition this checks on it would be slightly more reliable.

Reply:
good point!